packages/compass-utils/src/intercom-track.ts

```diff
@@ -0,0 +1,38 @@
+import { createLoggerAndTelemetry } from '@mongodb-js/compass-logging';
+const { debug } = createLoggerAndTelemetry('COMPASS-INTERCOM');
+
+type WindowWithIntercomGlobals = Window &
+  typeof globalThis & {
+    Intercom?: (...args: any[]) => any;
+    attachEvent?: (...args: any[]) => any;
+  };
+
+export enum IntercomTrackingEvent {
+  submittedNlPrompt = 'submitted-nl-prompt',
+}
+
+export function intercomTrack(
+  event: IntercomTrackingEvent,
+  metadata?: unknown
+) {
+  const win = window as WindowWithIntercomGlobals;
+  if (typeof win === 'undefined') {
+    return;
+  }
+
+  if (
+    !win.Intercom ||
+    typeof win.Intercom !== 'function' ||
+    // the last check is because the packages might be embedded in an environment where an external Intercom lives
+    // this way we ensure we don't polute such external Intercom with events intended for Compass Intercom
+    !process.env.HADRON_METRICS_INTERCOM_APP_ID
+  ) {
+    return;
+  }
+
+  try {
+    win.Intercom('track', event, metadata);
+  } catch (error) {
+    debug('intercom track error', error);
```

Comment:
Just FYI: I don't think debug() would end up in a log file. If you're interested in seeing this in logs if it happens you would have to use log() on it.

Reply:
hmm, I thought it'd make sense to use the same level as when the intercom fails to load -> https://github.com/mongodb-js/compass/blob/20a6ff86654818b70348e61ee611a29afade7277/packages/compass/src/app/intercom/intercom-script.ts#L37.
but tbh I'm not sure why that level was chosen in that instance either 🤔

Reply:
probably because it's not an user facing failure, so noone would look into the logs unless debugging anyway? or are there other cases where you'd imagine we'd need this in the logs?

Reply:
The problem with debug() is that you can only see it if you have debug enabled and you're looking at the console or stdout. So it isn't something a user can spot in a log file and send it to us and we'd have to already know how to recreate the intercom failure in order to generate any debug output locally.

Which might be fine. Just thought I'd bring it up.